whelktool/scripts/cleanups/2022/10/lxl-3756-move-carriertype.groovy

```diff
@@ -14,13 +16,17 @@ selectBySqlWhere(where) { data ->
         sourceList.removeElement(OBJECT_TO_MOVE)
         if(sourceList.isEmpty()) {
             sourceList = data.graph[1].hasPart[0].remove('carrierType')
-            if(data.graph[1].hasPart[0].isEmpty()) {
-                data.graph[1].hasPart.remove(0)
-                if(data.graph[1].hasPart.isEmpty()) {
-                    data.graph[1].remove('hasPart')
-                }
+        }
+        if(data.graph[1].hasPart[0][TYPE] == ELECTRONIC) {
+            data.graph[1].hasPart[0].remove(TYPE)
+        }
+        if(data.graph[1].hasPart[0].isEmpty()) {
```

Comment:
In case the part isn't empty when type is removed, it would become untyped (which can be problematic), so I'd suggest combining these, like:
```groovy
if(data.graph[1].hasPart[0].size() == 1 && data.graph[1].hasPart[0][TYPE] == ELECTRONIC) {
```